packages/plugin-coverage/src/lib/nx/coverage-paths.ts

```diff
@@ -0,0 +1,47 @@
+import type { ProjectGraphProjectNode } from '@nx/devkit';
+import { join } from 'node:path';
+import { CoverageResult } from '../config';
+
+/**
+ *
+ * @param coverageFolder root folder containing all coverage results
+ * @param targets nx targets to be used for measuring coverage, test by default
+ * @returns An array of coverage result information for the coverage plugin.
+ */
+export async function getNxCoveragePaths(
+  targets: string[] = ['test'],
+): Promise<CoverageResult[]> {
+  const { createProjectGraphAsync } = await import('@nx/devkit');
+  const { nodes } = await createProjectGraphAsync({ exitOnError: false });
+
+  const coverageResults = targets.map(target => {
+    const relevantNodes = Object.values(nodes).filter(graph =>
+      hasNxTarget(graph, target),
+    );
+
+    return relevantNodes
+      .map(node => node.data)
+      .map<CoverageResult>(projectConfig => {
+        const { reportsDirectory } = projectConfig.targets?.[target]
+          ?.options as {
+          reportsDirectory: string;
+        };
```

Comment:
Could you also check for [`coverageDirectory` used by `@nx/jest`](https://github.com/nrwl/nx/blob/master/packages/jest/src/executors/jest/schema.json#L189-L193)? :pray: Jest is the default testing framework in Nx, so it would be great if we could support it.